### PR TITLE
rambox: don't build from sources

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

See #76618

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @Ma27 @Mic92

### CODE REVIEW

**Review by lukateras — APPROVED** [review]
Sorry for the inconvenience!

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 1]
```diff
@@ -1,52 +1,45 @@
...
+    mkdir -p $out/bin $out/opt/RamboxPro $out/share/applications
+    asar e resources/app.asar $out/opt/RamboxPro/resources/app.asar.unpacked   
     ln -s ${desktopItem}/share/applications/* $out/share/applications
-    runHook postInstall
   '';
 
   postFixup = ''
-    makeWrapper ${electron}/bin/electron $out/bin/rambox \
-      --add-flags "${rambox-bare} --without-update" \
+    makeWrapper ${electron}/bin/electron $out/bin/ramboxpro \
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix]: Is it supposed to be called `ramboxpro` in both versions?

**Comment by Mic92** [comment]
The non-pro version does not start for me (also it is called ramboxpro atm):

$ ramboxpro
App threw an error during load
Error: EEXIST: file already exists, mkdir '/home/joerg/.config/Rambox'
    at Object.mkdirSync (fs.js:731:3)
    at make (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/make-dir/index.js:61:12)
    at Function.module.exports.sync (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/make-dir/index.js:84:9)
    at ElectronStore.get store [as store] (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/conf/index.js:141:13)
    at new Conf (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/conf/index.js:46:26)
    at new ElectronStore (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/electron-store/index.js:20:3)
    at Object.<anonymous> (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/electron/main.js:21:16)
    at Object.<anonymous> (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/electron/main.js:509:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
A JavaScript error occurred in the main process
Uncaught Exception:
Error: EEXIST: file already exists, mkdir '/home/joerg/.config/Rambox'
    at Object.mkdirSync (fs.js:731:3)
    at make (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/make-dir/index.js:61:12)
    at Function.module.exports.sync (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/make-dir/index.js:84:9)
    at ElectronStore.get store [as store] (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/conf/index.js:141:13)
    at new Conf (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/conf/index.js:46:26)
    at new ElectronStore (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/node_modules/electron-store/index.js:20:3)
    at Object.<anonymous> (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/electron/main.js:21:16)
    at Object.<anonymous> (/nix/store/dc6rvad9ln7nnm8nmi10qxnq5d8aybgk-rambox-0.7.3/opt/RamboxPro/resources/app.asar.unpacked/electron/main.js:509:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"

**Comment by gnidorah** [comment]
@Mic92 
> The non-pro version does not start for me (also it is called ramboxpro atm):

Looks like that solution doesn't work with rambox community edition, I've changed back to using binaries instead of launching electron

@yegortimoshenko No need to sorry, it was a nice idea, but I would like to simplify maintaining of rambox

**Comment by Mic92** [comment]
I had to delete ~/.config/Rambox and than it worked.

**Comment by gnidorah** [comment]
@Mic92 Thank you!

**Comment by Mic92** [comment]
We should probably backport this as we in future likely also need to upgrade rambox in stable as well to unbrick services.

**Review by Ma27 — COMMENTED** [review]
> Sorry for the inconvenience!

No worries! It was a great effort since source-builds should be preferred over pre-built binary blobs in general IMHO. Also, the error is pretty hard to catch and reproduce (already seen several hash-updates in this package that weren't reproducible by everyone).

@gnidorah the change seems mostly fine, I have two questions though:

* I stopped rambox 0.7.3 (from a fairly recent `release-20.03`) and ran `./result/bin/rambox` after building it on your branch. The program worked fine, however *all* of my tabs were gone. After stopping this rambox process and reopening the one I installed globally, the tabs reappeared. Do you know if this is somehow related to the environment `rambox` is started in or an actual regression of your patch? In case of the latter it would be awesome if you could either try to find a fix or document this in the release notes at least.

* The comment at the top of `fetchNodeModules.nix`says that `Only npm >= 5.4.2 is deterministic`. So is there any chance to provide another fix atm? (or is it known what caused the issues?).

**Comment by Ma27** [comment]
So I was obviously too slow to with writing a review :sweat_smile: 

@gnidorah would you mind taking a look at my comments anyway? :)

**Comment by gnidorah** [comment]
@Ma27
> I stopped rambox 0.7.3 (from a fairly recent release-20.03) and ran ./result/bin/rambox after building it on your branch. The program worked fine, however all of my tabs were gone. After stopping this rambox process and reopening the one I installed globally, the tabs reappeared. Do you know if this is somehow related to the environment rambox is started in or an actual regression of your patch? In case of the latter it would be awesome if you could either try to find a fix or document this in the release notes at least.

Can't reproduce. Do you use auth0?

**Comment by Mic92** [comment]
@gnidorah I might did that too.

**Pull request merged**